### PR TITLE
Better Kafka test

### LINKED ISSUE #132: Missing (Fixing) unit test for FeatureRowKafkaIO
Encountering the below issue when trying to run the unit test for FeatureRowKafkaIO

```
15:12:04 [direct-runner-worker] INFO  org.apache.kafka.common.utils.AppInfoParser - Kafka version : 2.0.0
15:12:04 [direct-runner-worker] INFO  org.apache.kafka.common.utils.AppInfoParser - Kafka commitId : 3402a8361b734732
15:12:04 [direct-runner-worker] INFO  org.apache.kafka.clients.Metadata - Cluster ID: BsCMKXsURGy8qHivvUcsFg
15:12:04 [direct-runner-worker] INFO  o.apache.kafka.clients.consumer.internals.Fetcher - [Consumer clientId=consumer-3, groupId=Reader-0_offset_consumer_1743545037_none] Resetting offset for partition test_topic-0 to offset 1.
15:12:04 [pool-13-thread-1] INFO  o.apache.kafka.clients.consumer.internals.Fetcher - [Consumer clientId=consumer-3, groupId=Reader-0_offset_consumer_1743545037_none] Resetting offset for partition test_topic-0 to offset 1.
15:12:05 [pool-13-thread-1] INFO  o.apache.kafka.clients.consumer.internals.Fetcher - [Consumer clientId=consumer-3, groupId=Reader-0_offset_consumer_1743545037_none] Resetting offset for partition test_topic-0 to offset 1.
15:12:06 [pool-13-thread-1] INFO  o.apache.kafka.clients.consumer.internals.Fetcher - [Consumer clientId=consumer-3, groupId=Reader-0_offset_consumer_1743545037_none] Resetting offset for partition test_topic-0 to offset 1.
15:12:07 [pool-13-thread-1] INFO  o.apache.kafka.clients.consumer.internals.Fetcher - [Consumer clientId=consumer-3, groupId=Reader-0_offset_consumer_1743545037_none] Resetting offset for partition test_topic-0 to offset 1.
15:12:08 [pool-13-thread-1] INFO  o.apache.kafka.clients.consumer.internals.Fetcher - [Consumer clientId=consumer-3, groupId=Reader-0_offset_consumer_1743545037_none] Resetting offset for partition test_topic-0 to offset 1.
15:12:09 [pool-13-thread-1] INFO  o.apache.kafka.clients.consumer.internals.Fetcher - [Consumer clientId=consumer-3, groupId=Reader-0_offset_consumer_1743545037_none] Resetting offset for partition test_topic-0 to offset 1.
15:12:10 [pool-13-thread-1] INFO  o.apache.kafka.clients.consumer.internals.Fetcher - [Consumer clientId=consumer-3, groupId=Reader-0_offset_consumer_1743545037_none] Resetting offset for partition test_topic-0 to offset 1.

```

The test just keep on running in an endless loop looging the above.

### PR DESCRIPTION
**Pull request by tims**
Create a kafka test that actually tests the source class
And remove the unnecessary FeatureRowKey proto that was only used by kafka, because Feast doesn't care what the keys are (they are just used by producers to dictate partitions).

Fixes #132

### CODE REVIEW

File: protos/feast/types/FeatureRow.proto [hunk 1]
```diff
@@ -25,12 +25,6 @@ option java_package = "feast.types";
 option java_outer_classname = "FeatureRowProto";
 option go_package = "github.com/gojek/feast/protos/generated/go/feast/types";
 
-message FeatureRowKey {
```
Comment (zhilingc) [inline protos/feast/types/FeatureRow.proto]: Will removing the `FeatureRowKey` affect how CoalesceRows functions?

Comment (tims) [reply protos/feast/types/FeatureRow.proto]: No, it's not used by anything except for the kafka source we never really needed it.

Coalesce always used it's own CoalesceKey proto defined in `ingestion/protoe/feast_ingestion/types/CoalesceKey.proto`.  It is internal to the ingestion project.

**Comment by tims** [comment]
> Just some concerns over removal of FeatureRowKey seeing that Feast ingestion is no longer strictly stateless

Does up comment above address your concern? If not can you elaborate?

File: protos/feast/types/FeatureRow.proto (hunk 1 above)
Comment (zhilingc) [reply protos/feast/types/FeatureRow.proto]: Ok. Just clarifying. It looks good to me.

**Pull request merged**